### PR TITLE
Reenable "Local" settings tab

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Re-enable the "Local" settings tab
  - It contains a single input to change the root directory of 
- Add settings icon to jump to "Local" settings tab, next to "Local" category

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

**Comment by somebody1234** [comment]
@MrFlashAccount the way this jumps to a specific settings tab is kinda janky... i think it's acceptable *short term* as a quick-n-dirty solution but we'd probably want something more robust in the long term

**Comment by somebody1234** [comment]
i guess ideally these would be routes? (and then we could have a single discriminated union with an exhaustive list of all routes for type-safety)

File: app/dashboard/src/layouts/CategorySwitcher.tsx [hunk 1]
```diff
@@ -284,8 +290,25 @@ export default function CategorySwitcher(props: CategorySwitcherProps) {
                   <div className="ml-[15px] mr-1 border-r border-primary/20" />
                   {element}
                 </div>
-              ) : (
+              ) : data.category !== Category.local ? (
                 element
+              ) : (
+                <div key={data.category} className="group flex items-center self-stretch">
+                  {element}
+                  <div className="grow" />
```
Comment (MrFlashAccount) [inline app/dashboard/src/layouts/CategorySwitcher.tsx]: why do we need this empty div here? it looks like we can put `ml-auto` on the button instead?

**Comment by MrFlashAccount** [comment]
> i guess ideally these would be routes? (and then we could have a single discriminated union with an exhaustive list of all routes for type-safety)

Yeah, routes make sense here. Just FYI, I don't remember if I mentioned it already or not, but when I tried to migrate to [`data router`](https://reactrouter.com/en/main/routers/picking-a-router#using-v64-data-apis), I found that the `react-router` doesn't satisfy our requirements and needs, but `tanstack-router` seems like a better option and provides a type-safety, as well as integration with react-aria-components

**Review by MrFlashAccount — APPROVED** [review]
Small nit found, other than that LGTM

File: app/dashboard/src/layouts/CategorySwitcher.tsx (hunk 1 above)
Comment (somebody1234) [reply app/dashboard/src/layouts/CategorySwitcher.tsx]: > it looks like we can put `ml-auto` on the button instead?

yeah i guess i'm just not a fan of `m*-auto` to get layouts to work\* - i'll try switching to `justify-between` first, and switch to `ml-auto` if that doesn't work

\* specifically, i feel like it makes the gap "belong" to the child element which i don't think feels right semantically

**Pull request merged**